rocks-lib/src/operations/install.rs

```diff
@@ -43,77 +51,118 @@ where
     result
 }
 
-#[async_recursion]
 async fn install_impl(
     progress: &MultiProgress,
     packages: Vec<(BuildBehaviour, PackageReq)>,
     pin: PinnedState,
     config: &Config,
     lockfile: &mut Lockfile,
 ) -> Result<Vec<LocalPackage>, InstallError> {
-    let mut result = Vec::new();
-    for (build_behaviour, package_req) in packages {
-        let bar = progress.add(ProgressBar::from(format!("💻 Installing {}", package_req)));
+    let bar = progress.add(ProgressBar::from(
+        "🔭 Resolving dependencies...".to_string(),
+    ));
+    let all_packages = join_all(packages.iter().map(|(behaviour, pkg)| {
+        flattened_install_specs(&bar, (*behaviour, pkg.clone()), &pin, config)
+    }))
+    .await
+    .into_iter()
+    .try_collect::<_, Vec<Vec<PackageInstallSpec>>, InstallError>()?
+    .into_iter()
+    .flatten()
+    .unique_by(|install_spec| install_spec.local_package_spec.id().clone())
+    .collect_vec();
```

Comment:
misc: I'm soo curious if this can be shortened down even further. It's like the code is taunting me to try and reduce it :joy:, this is fine for the time being.

Reply:
The reason this has so much boilerplate is that `join_all().await` returns a `Vec`, not an `Iterator` :disappointed: